src/test/java/mate/academy/dao/UserDaoTest.java

```diff
@@ -0,0 +1,89 @@
+package mate.academy.dao;
+
+import java.util.Optional;
+import java.util.Set;
+import mate.academy.dao.impl.RoleDaoImpl;
+import mate.academy.dao.impl.UserDaoImpl;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.model.Role;
+import mate.academy.model.User;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class UserDaoTest extends AbstractTest {
+    private static final String BOB_EMAIL = "bob@i.ua";
+    private static final String BOB_PASSWORD = "123";
+    private static final String ELLIS_EMAIL = "allis@i.ua";
+    private static final String ELLIS_PASSWORD = "123";
+    private static final String EMPTY_EMAIL = "";
+    private static final String NOT_EXIST_EMAIL = "skjdfnkjdsfn";
+    private static final Long CHECK_ID = 1L;
+    private static final Long NULL_ID = null;
```

Comment:
Do not create constants for **null** or **id**
You can have different id and you do not need to check exact value